File: python/gender_age_emotion.py
import requests
import json


# Detect all human faces present in a given image and
# try to guess their emotions, age and gender.


# Target image: Feel free to change to whatever image holding as many human faces as you want
img = 'http://www.scienceforums.com/uploads/1282315190/gallery_1625_35_9165.jpg'

req = requests.get('http://api.pixlab.io/facemotion',params={
	'img': img,
	'key':'PixLab_API_Key',
})
reply = req.json()
if reply['status'] != 200:
	print (reply['error'])
	exit();

total = len(reply['faces']) # Total detected faces
print(str(total)+" faces were detected")
# Extract each face now 
for face in reply['faces']:
	cord = face['rectangle']
	print ('Face coordinate: width: ' + str(cord['width']) + ' height: ' + str(cord['height']) + ' x: ' + str(cord['left']) +' y: ' + str(cord['top']))
	# Guess emotion
	for emotion in face['emotion']:
		if emotion['score'] > 0.5:
			print ("Emotion - "+emotion['state']+': '+str(emotion['score']))
	# Grab the age and gender
	print ("Age ~: " + str(face['age']))
	print ("Gender: " + str(face['gender']))

File: PHP/gender_age_emotion.php
<?php
/*
 * PixLab PHP Client which is just a single class PHP file without any dependency that you can get from Github
 * https://github.com/symisc/pixlab-php 
 */
require_once "pixlab.php";

# Detect all human faces present in a given image and try to guess their age, gender and emotion state via their facial shapes.
# https://pixlab.io/#/cmd?id=facemotion for more info.

# Target Image: Feel free to change to whatever image holding as many human faces as you want.
$img = 'http://www.scienceforums.com/uploads/1282315190/gallery_1625_35_9165.jpg';

# Your PixLab key
$key = 'My_Pixlab_Key';
/* Analyze */
$pix = new Pixlab($key);
if( !$pix->get('facemotion',array(
	'img' => $img
	)) ){
	echo $pix->get_error_message();
	die;
}
/* Grab the total number of detected faces */
$faces = $pix->json->faces;
echo "Total number of detected faces: ".count($faces)."\n";
if( count($faces) < 1 ){
	echo "No human faces were were detected on this picture\n";
}else{
	# Iterate all over the detected faces
	foreach($faces as $face){
		$coord = $face->rectangle;
		echo 'Face coordinate: width: ' . $coord->width . ' height: ' . $coord->height . ' X: ' . $coord->left . ' Y: ' . $coord->top . "\n";
		# Guess emotion
		foreach( $face->emotion as $emotion ){
			if ( $emotion->score > 0.5 ){
				echo "Emotion - " . $emotion->state. ': ' . $emotion->score. "\n";
			}
		}
		# Grab the age and gender
		echo "Age ~: " . $face->age. "\n";
		echo "Gender: " . $face->gender . "\n";
	}
}
?>
